Consolidate rest api

AScheuss opened the pull request:
Es  müssten noch mehr Tests gemacht werden...

AScheuss committed:
Introduce resource getChanged and consolidation of other methods

AScheuss committed:
Introduced controlled Exception Handling (for custom Exceptions)

AScheuss committed:
Member fields in config mL

AScheuss committed:
Added some tests for RestApiMember.php

cyrillbolliger commented on app/Exceptions/Handler.php:
```diff
@@ -46,6 +47,9 @@ public function report(Exception $exception)
     public function render($request, Exception $exception)
     {
+      if ($exception instanceof ClientException) {
+        abort(400, "Wrong db-key.");
```
Die ClientException könnte auch andere Ursachen haben, zB das der Webling Server down ist, etc. Wollen wir einfach die message weitergeben und prefixen, also zB `Webling Exception: $exception->getMessage()`?

AScheuss replied:
👍

cyrillbolliger commented on routes/api.php:
```diff
@@ -14,8 +14,26 @@
 */
 $version = 'v1';
+$adminBase=$version . '/admin';
+$memberBase=$version . '/member';
```
Diese Variabeln wäre im global scope, oder?

AScheuss replied:
Vom Prinzip her schon. Wo soll ich sie also hintun? (In der ".env"-Config machen sie aber wenig Sinn...)

cyrillbolliger replied:
Hmm, wusste ich auch nicht auf Anhieb. Bin vorher aber auf _Route Prefixing_ gestossen. Vielleicht wäre das eine elegante Variante? https://laravel.com/docs/5.0/routing#route-groups

cyrillbolliger commented on tests/Unit/Http/Controllers/RestApi/ApiHelperTest.php:
```diff
@@ -0,0 +1,66 @@
+      ApiHelper::checkIntegerInput('11');
+      ApiHelper::checkIntegerInput(11);
+
+      $this->expectException(IllegalArgumentException::class);
```
PHPUnit ist leider doof genug, dass auch wenn einer der ersten `ApiHelper::checkIntegerInput(11)` fehlschlagen würde, es einen `success` gibt, da das `expectException` funktionsweit gilt...

AScheuss replied:
Ok, werde den Test anpassen

cyrillbolliger reviewed (approved):
Sehr nice und clean. 1000 Dank :+1:

AScheuss committed:
Use Routing Groups

AScheuss committed:
Consolidate ApiHelperTests

No test of createMemberRepo because the logic seems to be too simple
(would need to be tested in MemberRepository.__construct)

AScheuss committed:
Add Tests for Exception Handling

Could be expanded

cyrillbolliger committed:
fix: test interference of field factory

AScheuss committed:
Add private  field in ApiHelperTest
